### DISCUSSION TITLE
how to change one master node's IP？

### DISCUSSION THREAD

**Discussion by GoodJoey** [k3s-io/k3s#5994] [General]
I've created a k3s cluster with 3 master nodes (embeded database: etcd), and one of the master node's IP needed to change its IP (IP conflict with other server).
Now, the master IP has been changed , and it can't be added back to the cluster. Although the k3s is still running well with two master nodes,  I'd like to add the master node back to the cluster with the new IP, is there any convenient way to do it?

**Comment by brandond** [comment]
IPs are expected to remain immutable for the life of a node's membership in the cluster. If you need to change the IP, you should stop the k3s service on the affected node, run `kubectl delete node` to remove the node from the cluster, and then rejoin the node.

**Reply by GoodJoey** [reply]
you are right. it's ok if it is a worker node.
In my case, the problem is that there are only two master left, i want to rejoin the node as a master, but i can't manage to do that.

**Reply by brandond** [reply]
Make sure the node is deleted from the cluster before rejoining it. You should be able to operate with only two servers.
